tests/unit/test_pqtools.py

```diff
@@ -0,0 +1,69 @@
+"""Tests for pqtools.py"""
+
+import os
+import contextlib
+from pathlib import Path
+from typing import Tuple
+
+import pytest
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+from zolltools.db import pqtools
+
+
+@contextlib.contextmanager
+def _temporary_parquet_table_context_manager(frame: pd.DataFrame) -> Path:
+    """
+    Context manager for a temporary parquet table (used for testing). The
+    context manager will delete the temporary directory and table upon closing.
+
+    :param frame: the data frame to make into a parquet table
+    :returns: the path to the temporary table
+    """
+
+    table_id = str(pd.util.hash_pandas_object(frame).sum() % 1_000_000).zfill(6)
```

Comment:
Why `1,000,000`? Maybe it can be set to match the max file name length?

Reply:
[It seems that Windows ppl are limited to 260 character paths](https://stackoverflow.com/questions/265769/maximum-filename-length-in-ntfs-windows-xp-and-windows-vista). Keeping the file name shorter seems to be polite. I think 6 characters is an appropriate amount of hashing.